### ISSUE TITLE
Bullet list firefox issue

### ISSUE THREAD

**Issue by xitanggg** [xitanggg/open-resume#15]
Documenting a feedback
> One thing I noticed was the really weird (I am on Firefox) : When I tried to enter text in the additional box and bullet points were activated it would create a bullet point after every space I pressed....

**Comment by mahalel** [comment]
Hey great tool. Can confirm I also have this issue on Firefox Desktop 114.0.2 (64-bit)

**Comment by DatSudo** [comment]
Up!

In bullet point sections, after the first word and a space, every letter goes to a new bullet point:

![image](https://github.com/xitanggg/open-resume/assets/76833632/2c6f203a-5f03-48b4-ba6b-b5499a7e6b47)

It's fine in Chrome though.

**Comment by xitanggg** [comment]
Apologize for this bug, I will prioritize fixing this issue early this week

**Comment by mahalel** [comment]
Please don't apologize, we appreciate your work and your time.

**Comment by Nicceboy** [comment]
Feels like there is some kind of counter issue? You can eventually get the text (full phrase) fit for the one bullet if you play around enough with spaces and backspace.

**Comment by williamtwild** [comment]
Can confirm this happens on Safafi Version 16.5 (18615.2.9.11.4) as well.

**Comment by xitanggg** [comment]
Just got a chance to take a closer look at the issue. (BulletListTextarea)[https://github.com/xitanggg/open-resume/blob/main/src/app/components/ResumeForm/Form/InputGroup.tsx#L115] is the component to manage the descriptions input. It is essentially a content editable div and uses `innerText` to set what user enters. A weird thing about firefox is that if it is at the end of the line and user presses space, `innerText` appends a new line character instead of appending a space as in other browser (Chrome, Edge). 

It is not as a simple fix as I would have thought. I just created an issue with Firefox on https://github.com/webcompat/web-bugs/issues/124217 to learn more about why `innerText` behaves such a way 🤔

**Comment by williamtwild** [comment]
What happens using innerHTML?

**Comment by xitanggg** [comment]
Thanks all for reporting this issue! I just created and released a fallback component `BulletListTextareaFallback` for Firefox and Safari as a workaround solution. Can anyone help test it on their end and confirm if it indeed solves the issue? 

The fallback component should work fine most of the case. There is a small edgecase I notice where it would refocus to the end if you have say 2 bullet points, and you try to backspace/remove the second bullet point to combine it with the first one. Hopefully this isn't common.

I initially created a custom `BulletListTextarea` component since I want the output to be an array of string type `descriptions: sting []` given that is the input format consumed by the react pdf render to render the description list. I used `innerText` since it would be easier to serialize and deserialize things by splitting the string by new line. Using `innerHTML` might require more work to split by `<div> <br>` and we might go into the same problem, as I notice Firefox adds a `<br>` in `innerHTML` if user adds a space at the end of the line.
![image](https://github.com/xitanggg/open-resume/assets/23644661/2093d835-a3d7-4daa-9e6d-1ff5c437b462)
My lesson is that each browser handles `innerText` and `innerHTML` very differently, and it can be risky to rely on them.
